### DISCUSSION TITLE
Get fields as type

### DISCUSSION THREAD

**Discussion by Darkside73** [noam-honig/radweb#416] [Q&A] [answered]
Is it possible to get entity's fields as type? For example:
```ts
@Entity('posts')
export class Post {
  @Fields.cuid()
    id = ''

  @Fields.string()
    title = ''
}

type PostFields = ExtractFields<Post>
```
Produces:
```ts
type PostFields = {
  id: string
  title: string
}
```

**Comment by noam-honig** [comment, marked as answer]
You can achieve what you want with:
```ts
export class Post {
    id = ''
    title = ''
}
type PostFields = InstanceType<typeof Post>;
```
But that's actually redundant, since in typescript a class is also a type - so the type `Post` has these exact same definition as the `PostFields` type created here.

so:
```ts
let x: Post = {
   id: '1',
   title: 'noam'
}
//is identical to:
let y: PostFields = {
  id: '1',
  title: 'noam'
}

```
Checkout this typescript playground
https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwDYEMDOa4AUJvgbwCg44BLAEzgF44ByW4uGUmJYauhgX0JgE8w7HHgBipYEnKYaASQB2eFHITAAKgOAAefoIgAzbLhgA+ANyFCbeCABchvByIkKd2gEZaAGiYs2ruRAoALbclsDwfHbCMGISUo6MLnQenozMrMD+gSGEPEA
